### PR TITLE
add new pollingInterval & cooldownPeriod tests

### LINKED ISSUE #4266: Add e2e test for SO updates of pollingInterval
### Proposal

create tests in `internals/` for testing different variations of `pollingInterval` and if SO updates during scaling works

### Use-Case

_No response_

### Is this a feature you are interested in implementing yourself?

Yes

### Anything else?

_No response_

### LINKED ISSUE #4267: Add e2e test for SO updates of cooldownPeriod
### Proposal

tests in `internals/` regarding `cooldownPeriod` updates during scaling

### Use-Case

_No response_

### Is this a feature you are interested in implementing yourself?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by gauron99**
- New `internals/` tests for PI & CdP
- I also created new helper.go function thats exactly same as `kubectl apply` but is `kubectl create` instead because of the job for updating metrics - when using apply, the test runs so fast that the job doesnt have time to get created, finished and deleted before the next apply and returns an error(job immutable). Because of this i used generated Job names as well.
- Sometimes there is a timing issue with `AssertReplicaCountNotChangeDuringTimePeriod` that the value changes before the 60 seconds (like 59s) therefore i added 2 seconds to the cooldowns to ensure it passes.
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4266 
Fixes #4267

### CODE REVIEW

File: tests/internals/polling_cooldown_so/polling_cooldown_so_test.go [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  ports:
+  - port: 8080
+    targetPort: 8080
+`
+)
+
+// use KubectlCreateWithTemplate() function because applying yaml template
+// for update-metrics relies on the previous job to be deleted
+// (job is immutable), but in this case the test runs too fast at some
+// points that job is not finished & deleted in time
```
Comment (JorTurFer) [inline tests/internals/polling_cooldown_so/polling_cooldown_so_test.go:197]: Isn't`KubectlApplyWithTemplate` with `ttlSecondsAfterFinished: 0` enough for removing the job?

File: tests/internals/polling_cooldown_so/polling_cooldown_so_test.go [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+	TriggerAuthName             string
+	SecretName                  string
+	ServiceName                 string
+	MetricsServerDeploymentName string
+	MetricsServerEndpoint       string
+	MinReplicas                 string
+	MaxReplicas                 string
+	MetricValue                 int
+	PollingInterval             string
+	CooldownPeriod              string
```
Comment (JorTurFer) [inline tests/internals/polling_cooldown_so/polling_cooldown_so_test.go:52]: Can we use `int` instead of `string`?

**Review by JorTurFer — COMMENTED** [review]
LGTM, some comments inline

File: tests/internals/polling_cooldown_so/polling_cooldown_so_test.go (hunk 1 above)
Comment (gauron99) [reply tests/internals/polling_cooldown_so/polling_cooldown_so_test.go:197]: its not, the job has a few seconds to run and it didnt finish in time.

**Comment by gauron99** [comment]
i changed string->int as suggested and also changed reserve time to 5 seconds because one of the times of my testing it still failed, so to avoid random failed runs, 5 sec should be enough

**Review by JorTurFer — APPROVED** [review]
LGTM!

**Pull request merged**